src/resolvers/User/posts.ts

```diff
@@ -0,0 +1,93 @@
+import { Post } from "../../models";
+import type {
+  ConnectionPageInfo,
+  UserResolvers,
+} from "../../types/generatedGraphQLTypes";
+import { graphqlConnectionFactory } from "../../utilities/graphqlConnectionFactory";
+import { parseRelayConnectionArguments } from "../../utilities/parseRelayConnectionArguments";
+
+/**
+ * Checks if a given cursor corresponds to a valid database object.
+ * @param cursor - The cursor to check.
+ * @returns A Promise that resolves to true if the cursor corresponds to a valid database object, otherwise false.
+ */
+export const isValidCursor = async (
+  cursor: string | null | undefined
+): Promise<boolean> => {
+  if (cursor == null) {
+    return false;
+  }
+  const result = await Post.findById(cursor);
+  return result ? true : false;
+};
+
+/**
+ * Resolver function to fetch and return posts created by a user from the database.
+ * @param parent - An object that is the return value of the resolver for this field's parent.
+ * @param args - Arguments passed to the resolver.
+ * @returns An object containing an array of posts,totalCount of post and pagination information.
+ */
+export const posts: UserResolvers["posts"] = async (parent, args) => {
+  const paginationArgs = parseRelayConnectionArguments(args, 10);
+
+  // If the cursor is not a valid database object, return default GraphQL connection.
+  if (!isValidCursor(paginationArgs.cursor)) {
+    return graphqlConnectionFactory();
```

Comment:
the point was to create the graphql connection at the beginning of resolver, and mutate accordingly it as the logic traverses down
```typescript
const postsConnection = graphqlConnectionFactory<InterfacePost>();
```
if the cursor is null you have to ignore this check, also no point seperating it out in a seperate function
```typescript
if(paginationArgs.cursor !== null) {
  const post = await Post.findById(cursor);
  if (!post) {
    return postsConnection
  }
}
```